File: lab-3/c#/Vector.cs
using System;

public class Vector
{
    private double length;
    private double angle;

    public Vector()
    {
        length = 0;
        angle = 0;
    }

    public Vector(double l, double a)
    {
        length = l;
        angle = a;
    }

    public Vector(Vector other)
    {
        length = other.length;
        angle = other.angle;
    }

    public void Rotate(double angle)
    {
        this.angle += angle;
    }

    public void Rotate()
    {
        this.angle += Math.PI / 2;
    }

    public double GetLength()
    {
        return length;
    }

    public double GetAngle()
    {
        return angle;
    }

    public static Vector operator +(Vector v1, Vector v2)
    {
        double x1 = v1.length * Math.Cos(v1.angle);
        double y1 = v1.length * Math.Sin(v1.angle);

        double x2 = v2.length * Math.Cos(v2.angle);
        double y2 = v2.length * Math.Sin(v2.angle);

        double x = x1 + x2;
        double y = y1 + y2;

        double newLength = Math.Sqrt(x * x + y * y);
        double newAngle = Math.Atan2(y, x);

        return new Vector(newLength, newAngle);
    }

    public static Vector operator *(Vector v, double scalar)
    {
        return new Vector(v.length * scalar, v.angle);
    }

    public static Vector AddVectors(params Vector[] vectors)
    {
        Vector result = new Vector();

        foreach (var v in vectors)
        {
            result = result + v;
        }

        return result;
    }

    public void Print()
    {
        Console.WriteLine($"Length: {length}, Angle: {angle} radians");
    }
}

File: lab-3/c++/Vector.cpp
#include "Vector.h"
#include <cmath>

Vector2D::Vector2D() : length(0), angle(0) {}

Vector2D::Vector2D(double l, double a) : length(l), angle(a) {}

Vector2D::Vector2D(const Vector2D& other) : length(other.length), angle(other.angle) {}

void Vector2D::rotate(double angle) {
    this->angle += angle;
}

double Vector2D::getLength() const {
    return length;
}

double Vector2D::getAngle() const {
    return angle;
}

Vector2D Vector2D::operator+(const Vector2D& other) const {
    double x1 = length * cos(angle);
    double y1 = length * sin(angle);

    double x2 = other.length * cos(other.angle);
    double y2 = other.length * sin(other.angle);

    double x = x1 + x2;
    double y = y1 + y2;

    double newLength = sqrt(x * x + y * y);
    double newAngle = atan2(y, x);

    return Vector2D(newLength, newAngle);
}

Vector2D Vector2D::operator*(double scalar) const {
    return Vector2D(length * scalar, angle);
}

void Vector2D::print() const {
    std::cout << "Length: " << length << ", Angle: " << angle << " radians" << std::endl;
}